flask/app/users.py

```diff
@@ -102,9 +88,80 @@ def reset_minio_user(username: str):
         # we can guarantee from POST /api/groups that the policy exists in MinIO
         minio_admin.group_add(group.group_code, access_key)
         minio_admin.set_policy(group.group_code, group=group.group_code)
-
     user.minio_access_key = access_key
     user.minio_secret_key = secret_key
+
+
+@app.route("/api/users/<string:username>", methods=["POST"])
+@login_required
+def reset_minio_user(username: str):
+    if (
+        not app.config.get("LOGIN_DISABLED")
+        and not current_user.is_admin
+        and username != current_user.username
+    ):
+        return "Unauthorized", 401
+    if not request.json:
+        return "Request body must be JSON", 415
+    user = (
+        models.User.query.options(joinedload(models.User.groups))
+        .filter(models.User.username == username)
+        .first_or_404()
+    )
+    # TODO: maybe rate limit this API because generating these is expensive
+    reset_minio_credentials(user)
     transaction_or_abort(db.session.commit)
+    return jsonify(
+        {
+            "minio_access_key": user.minio_access_key,
+            "minio_secret_key": user.minio_secret_key,
+        }
+    )
+
 
-    return jsonify({"minio_access_key": access_key, "minio_secret_key": secret_key})
+def valid_strings(body: Dict[str, Any], *keys: str) -> bool:
+    return all(
+        map(
+            lambda key: key in body and isinstance(body[key], str) and len(body[key]),
+            keys,
+        )
+    )
+
+
+@app.route("/api/users", methods=["POST"])
+@login_required
+@check_admin
+def create_user():
+    if not request.json:
+        return "Request body must be JSON", 415
+
+    if not valid_strings(request.json, "username", "email", "password"):
+        return "Missing fields", 400
+
+    user = models.User.query.filter_by(username=request.json["username"]).first()
+    if user is not None:
+        return "User already exists", 422, {"location": f"/api/users/{user.username}"}
+    user = models.User.query.filter_by(email=request.json["email"]).first()
+    if user is not None:
+        return "Email already used", 422, {"location": f"/api/users/{user.username}"}
```

Comment:
Optional: Could be done nicer using `|` similar to https://github.com/ccmbioinfo/ST2020/blob/ec00df4054838d05a4bbc1d197acdaea052d04e2/flask/app/groups.py#L143